File: chap15_muscle/exercises/assay.py
from sympy import *

V0,V1,R = symbols("V0 V1 R")

V = V0 + V1*R
pp = 1 - (2*V/R)*(1 - exp(-R/(2*V)))*(1+V*R)
out = Poly(series(pp,R,0,3).removeO(),R)

cc = out.coeffs()
print(cc)
V0val = solve(cc[1],V0)[1]
print(V0val)

dum = cc[0].subs(V0,V0val)
print(solve(dum,V1))

File: chap15_muscle/exercises/assay.m
function assay

clear all
close all
clc

syms R V0 V1 V2

V = V0 + R*V1;
ff = taylor(1/V, R, 'Order', 4);
ee = taylor(1-exp(-R/(2*V)),R,'Order',4);

wun = 1 - 2*V/R*ee*(1+R*V);
cc = collect(simplify(wun),R);
cf = coeffs(cc,R);
cf1 = simplify(cf(1))
V0sol = solve(cf1==0,V0)
cf2 = simplify(subs(cf(2),V0,V0sol))
V1sol = solve(cf2==0,V1)

end